ext/rotoscope/rotoscope.c

```diff
@@ -156,12 +158,15 @@ static rs_tracepoint_t extract_full_tracevals(rb_trace_arg_t *trace_arg, const r
   rs_class_desc_t method_owner = tracearg_class(trace_arg);
   rb_event_flag_t event_flag = rb_tracearg_event_flag(trace_arg);
 
+  VALUE method_name = tracearg_method_name(trace_arg);
+  VALUE filepath = callsite->filepath;
+
   return (rs_tracepoint_t){
       .event = evflag2name(event_flag),
-      .entity = method_owner.name,
-      .method_name = tracearg_method_name(trace_arg),
+      .entity = StringValueCStr(method_owner.name),
+      .method_name = StringValueCStr(method_name),
       .method_level = method_owner.method_level,
-      .filepath = callsite->filepath,
+      .filepath = StringValueCStr(filepath),
       .lineno = callsite->lineno};
```

Comment:
I decided to keep Ruby `VALUE` types out of `rs_tracepoint_t`, since nothing needs them in that form. Instead I keep the values as null-terminated strings 👍